docs/partition-guide.md

```diff
@@ -0,0 +1,124 @@
+<!--
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+
+### CarbonData Partition Table Guidance
+This guidance illustrates how to create & use partition table in CarbonData.
+
+* [Create Partition Table](#create-partition-table)
+  - [Create Hash Partition Table](#create-hash-partition-table)
+  - [Create Range Partition Table](#create-range-partition-table)
+  - [Create List Partition Table](#create-list-partition-table)
+* [Show Partitions](#show-partitions)
+* [Maintain the Partitions](#maintain-the-partitions)
+* [Partition Id](#partition-id)
+* [Tips](#tips)
+
+### Create Partition Table
+
+##### Create Hash Partition Table
+```
+   CREATE TABLE [IF NOT EXISTS] [db_name.]table_name
+                    [(col_name data_type , ...)]
+   STORED BY 'carbondata'
+   PARTITIONED BY (partition_col_name data_type)
```

Comment:
The ‘PARTITIONED BY' must be before 'STORED BY', otherwise it will throw an error:
`mismatched input 'PARTITIONED' expecting {<EOF>, '(', 'SELECT', 'FROM', 'AS', 'WITH', 'VALUES', 'TABLE', 'INSERT', 'MAP', 'REDUCE', 'TBLPROPERTIES', 'LOCATION'}(line 11, pos 2)`

Reply:
I can see there are two functions validateTypeConvert and validateTypeConvertForSpark2 in CommonUtil.scala to validate partition column type. It's added by some other contributor.